dashboard/app/controllers/user_scripts_controller.rb

```diff
@@ -0,0 +1,19 @@
+class UserScriptsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_user_script, only: :update
```

Comment:
Why not just call `load_and_authorize_resource`?

The new [ability](https://github.com/code-dot-org/code-dot-org/pull/23720/files#diff-38d6349150d685325a41324b3032884eR64) already scopes UserScript by user id, so all you need is

```ruby
load_and_authorize_resource id_param: :script_id
```

(See comments here for [options](https://github.com/CanCanCommunity/cancancan/blob/develop/lib/cancan/controller_additions.rb#L17-L121))

Reply:
Does `load_and_authorize_resource` provide the `find_or_create_by` behavior implemented in `set_user_script` below?

Reply:
Ah, no. I didn't notice that. Thanks.

Ok, if we want to find or create, we need to do it manually, but let's at least call `authorize_resource` instead of `authenticate_user!`. Otherwise why bother creating the ability?

Reply:
based on this exchange, I've removed the new ability.